### PR TITLE
Renaming components for clarity

### PR DESCRIPTION
**Pull request by tanveergill**
### Description of change
* Renames:
  * ConcurrencyLimiter -> LoadScheduler
  * AIMDConcurrencyLimiter -> AdaptiveLoadScheduler
  * LoadRegulator -> Regulator
  * LoadShaper -> LoadRamp
* Initial changes towards load shed confirmatory criteria

##### Checklist

- [x] Tested in playground or other setup
- [x] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added

Screenshot from 2023-04-27 00-26-10

### Summary by OpenAI

**New Feature:**
- Added flow control components for improved concurrency and rate limiting.
- Introduced new proto definitions for policy synchronization.
- Implemented AIMD concurrency controller and load regulator components.
- Updated blueprints and policies for feature rollout and latency-based concurrency limiting.

**Documentation:**
- Enhanced documentation for flow control components, bundled blueprints, tutorials, and observability.

> 🎉 Flow control's here to stay, 🚀
> Concurrency and rate limiting lead the way. 🌟
> With new components and blueprints refined, 🛠️
> Our system's performance is now well-aligned! 🏆

### CODE REVIEW

**Review by harjotgill — APPROVED** [review]

File: docs/content/tutorials/auto-scale/load-based-auto-scaling/load-based-auto-scaling.md [hunk 1]
```diff
@@ -28,8 +28,8 @@ configured using _Controllers_ to adjust the number of instances allocated to
 the service. Load-based auto-scaling is achieved by defining a scale-out
 _Controller_ that acts on a load-shedding signal (load multiplier) signal from
 the blueprint. This signal measures the fraction of traffic that the
-[_Concurrency Limiter_](concepts/flow-control/components/concurrency-limiter.md)
-is shedding. The _Auto Scaler_ is configured to scale-out using a _Gradient
+[_Load Scheduler_](concepts/flow-control/components/load-scheduler.md) is
+shedding. The _Auto Scaler_ is configured to scale-out using a _Gradient
 Controller_ based on this signal and a setpoint of 1.0.
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/tutorials/auto-scale/load-based-auto-scaling/load-based-auto-scaling.md:33]: It is already the new link

File: blueprints/policies/feature-rollout/base/policy.libsonnet [hunk 2]
```diff
@@ -53,7 +54,7 @@ function(cfg) {
     reset_signals: driverAccumulator.reset_signals + (if isReset then [resetSignal] else []),
     forward_signals_count: driverAccumulator.forward_signals_count + if isForward then 1 else 0,
     backward_signals_count: driverAccumulator.backward_signals_count + if isBackward then 1 else 0,
-    reset_signals_count: driverAccumulator.reset_signals_count + if driver.reset != null then 1 else 0,
+    reset_signals_count: driverAccumulator.reset_signals_count + if criteria.reset != null then 1 else 0,
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply blueprints/policies/feature-rollout/base/policy.libsonnet]: Thanks, fixed

File: blueprints/policies/feature-rollout/base/policy.libsonnet [hunk 3]
```diff
@@ -174,7 +177,7 @@ function(cfg) {
...
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply blueprints/policies/feature-rollout/base/policy.libsonnet]: Thanks, fixed

File: docs/content/reference/policies/bundled-blueprints/policies/feature-rollout/base.md [hunk 4]
```diff
@@ -643,7 +667,7 @@
     description='List of promql drivers that compare results of a Prometheus query against forward, backward and reset thresholds.'
     type='Array of Object (promql_driver)'
     reference='#promql-driver'
-    value='[{"backward": {"operator": "__REQUIRED_FIELD__", "threshold": "__REQUIRED_FIELD__"}, "forward": {"operator": "__REQUIRED_FIELD__", "threshold": "__REQUIRED_FIELD__"}, "query_string": "__REQUIRED_FIELD__", "reset": {"operator": "__REQUIRED_FIELD__", "threshold": "__REQUIRED_FIELD__"}}]'
+    value='[{"criteria": {"backward": {"operator": "__REQUIRED_FIELD__", "threshold": "__REQUIRED_FIELD__"}, "forward": {"operator": "__REQUIRED_FIELD__", "threshold": "__REQUIRED_FIELD__"}, "reset": {"operator": "__REQUIRED_FIELD__", "threshold": "__REQUIRED_FIELD__"}}, "query_string": "__REQUIRED_FIELD__"}]'
 />
 
 <!-- vale on -->
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/policies/bundled-blueprints/policies/feature-rollout/base.md:672]: Confirmed that it looks okay in the blueprints reference doc

File: docs/content/reference/observability/flow-metrics/flow-metrics.md [hunk 5]
```diff
@@ -18,25 +18,27 @@
...
+| aperture.source                   | single      | sdk, envoy                                                                                                                                                                | Aperture Flow source                               | SDKs, Envoy               |
+| aperture.decision_type            | single      | DECISION_TYPE_ACCEPTED, DECISION_TYPE_REJECTED                                                                                                                            | Decision type taken by policy                      | SDKs, Envoy               |
+| aperture.reject_reason            | single      | REJECT_REASON_NONE, REJECT_REASON_RATE_LIMITED, REJECT_REASON_LOAD_SHED                                                                                                   | Reject reason of the decision taken by policy      | SDKs, Envoy               |
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/observability/flow-metrics/flow-metrics.md:28]: Okay

File: docs/content/reference/observability/flow-metrics/flow-metrics.md [hunk 6]
```diff
@@ -18,25 +18,27 @@
...
+| aperture.dropping_rate_limiters   | multi-value | "policy_name:s1, component_id:18, policy_hash:5kZjj"                                                                                                                      | Rate limiters dropping the traffic                 | SDKs, Envoy               |
+| aperture.load_scheduler           | multi-value | "policy_name:s1, component_id:13, policy_hash:5kZjj"                                                                                                                      | Concurrency limiters matched to the traffic        | SDKs, Envoy               |
+| aperture.dropping_load_scheduler  | multi-value | "policy_name:s1, component_id:13, policy_hash:5kZjj"                                                                                                                      | Concurrency limiters dropping the traffic          | SDKs, Envoy               |
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/observability/flow-metrics/flow-metrics.md:32]: That's what the new component naming convention is. The OLAP schema has to be consistent with the component names that matched and caused flow to be dropped

File: pkg/policies/flowcontrol/engine_test.go [hunk 7]
```diff
@@ -71,25 +71,25 @@ var _ = Describe("Dataplane Engine", func() {
...
 		})
 
 		It("Unregisters existing scheduler actuator", func() {
-			err := engine.RegisterConcurrencyLimiter(mockConLimiter)
-			err2 := engine.UnregisterConcurrencyLimiter(mockConLimiter)
+			err := engine.RegisterLoadScheduler(mockConLimiter)
+			err2 := engine.UnregisterLoadScheduler(mockConLimiter)
 			Expect(err).NotTo(HaveOccurred())
 			Expect(err2).NotTo(HaveOccurred())
 		})
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply pkg/policies/flowcontrol/engine_test.go:95]: Okay

File: pkg/policies/flowcontrol/engine_test.go [hunk 8]
```diff
@@ -161,28 +161,28 @@ var _ = Describe("Dataplane Engine", func() {
...
 			svcs := []string{"testService.testNamespace.svc.cluster.local"}
 			labels := map[string]string{"service": "testService.testNamespace.svc.cluster.local"}
 
 			mmr := engine.(*Engine).getMatches(controlPoint, svcs, labels)
 			Expect(mmr.fluxMeters).NotTo(BeEmpty())
-			Expect(mmr.concurrencyLimiters).NotTo(BeEmpty())
+			Expect(mmr.loadSchedulers).NotTo(BeEmpty())
 		})
 	})
 })
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply pkg/policies/flowcontrol/engine_test.go:188]: Okay

File: docs/content/reference/policies/spec.md [hunk 9]
```diff
@@ -166,6 +166,18 @@
...
+<!-- vale off -->
+
+([InPort](#in-port))
+
+<!-- vale on -->
+
+The enabled port controls whether the _Load Scheduler_ can load shed _Flows_. By default, the _Load Scheduler_ is enabled.
+
+</dd>
 <dt>setpoint</dt>
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/policies/spec.md:181]: Okay

File: docs/content/reference/policies/spec.md [hunk 10]
```diff
@@ -1486,12 +1755,6 @@
...
 
-:::info
-
-See also [_Concurrency Limiter_ overview](/concepts/flow-control/components/concurrency-limiter.md).
-
-:::
-
 It's based on the actuation strategy (for example, load actuator) and workload scheduling
 which is based on Weighted Fair Queuing principles.
 Concurrency is calculated in terms of total tokens per second, which can translate
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/policies/spec.md:1760]: This is still there for backward compatibility reasons

File: docs/content/reference/policies/spec.md [hunk 11]
```diff
@@ -2568,7 +2882,8 @@
 
 <!-- vale on -->
 
-LoadShaper is a component that shapes the load of the service.
+_Load Shaper_ is a component that shapes the load at a _Control Point_.
+Deprecated: 1.6.0
 
 </dd>
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/policies/spec.md]: There for backward compatibility reasons

File: docs/content/reference/policies/bundled-blueprints/policies/feature-rollout/base.md [hunk 12]
```diff
@@ -706,13 +730,13 @@
...
-    name='load_shaper'
-    description='Default configuration for flow regulator that can be updated at the runtime without shutting down the policy.'
-    type='Object (aperture.spec.v1.FlowRegulatorDynamicConfig)'
-    reference='../../../spec#flow-regulator-dynamic-config'
+    name='regulator'
+    description='Default configuration for load regulator that can be updated at the runtime without shutting down the policy.'
+    type='Object (aperture.spec.v1.RegulatorDynamicConfig)'
+    reference='../../../spec#regulator-dynamic-config'
     value='"__REQUIRED_FIELD__"'
 />
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/policies/bundled-blueprints/policies/feature-rollout/base.md:741]: Okay

File: docs/content/reference/policies/spec.md [hunk 13]
```diff
@@ -249,6 +261,7 @@
 <!-- vale on -->
 
 Is overload is a Boolean signal that indicates whether the service is overloaded based on the deviation of the signal from the setpoint taking into account some tolerance.
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/policies/spec.md:263]: Improved docs

File: docs/content/reference/policies/spec.md [hunk 14]
```diff
@@ -166,6 +166,18 @@
...
+<dt>enabled</dt>
+<dd>
+
+<!-- vale off -->
+
+([InPort](#in-port))
+
+<!-- vale on -->
+
+The enabled port controls whether the _Load Scheduler_ can load shed _Flows_. By default, the _Load Scheduler_ is enabled.
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply docs/content/reference/policies/spec.md:178]: Improved docs

File: pkg/policies/controlplane/circuitfactory/flow-control-components.go [hunk 15]
```diff
@@ -26,24 +28,158 @@ func newFlowControlCompositeAndOptions(
...
 
 		// Scheduler
-		if schedulerProto := concurrencyLimiterProto.GetScheduler(); schedulerProto != nil {
+		if schedulerProto := loadSchedulerProto.GetScheduler(); schedulerProto != nil {
 			// Use the same id as the component stack since agent sees only the component stack and generates metrics tagged with the component stack id
-			scheduler, schedulerOptions, err := concurrency.NewSchedulerAndOptions(schedulerProto, componentID.String(), policyReadAPI, agentGroupName)
+			scheduler, schedulerOptions, err := loadscheduler.NewSchedulerAndOptions(schedulerProto, componentID.String(), policyReadAPI, agentGroupName)
 			if err != nil {
 				return retErr(err)
 			}
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply pkg/policies/controlplane/circuitfactory/flow-control-components.go:185]: Incorporated an improved version of this. Fix on the way

File: pkg/policies/controlplane/circuitfactory/component-factory.go [hunk 16]
```diff
@@ -99,7 +100,21 @@ func NewComponentAndOptions(
...
+			err = jsonpb.UnmarshalString(jsonStr, loadRegulatorProto)
+			if err != nil {
+				return Tree{}, nil, nil, fmt.Errorf("error unmarshaling JSON to LoadRegulator: %v", err)
+			}
+			ctor = mkCtor(loadRegulatorProto, loadregulator.NewLoadRegulatorAndOptions)
+		case *policylangv1.FlowControl_LoadRegulator:
+			ctor = mkCtor(flowControlConfig.LoadRegulator, loadregulator.NewLoadRegulatorAndOptions)
 		default:
 			return newFlowControlCompositeAndOptions(flowControl, componentID, policyReadAPI)
 		}
```
[the remark this answers is not in the document]
Comment (tanveergill) [reply pkg/policies/controlplane/circuitfactory/component-factory.go]: Will be using `convertOldComponentToNew` instead

**Pull request merged**